archive/indexer.go

```diff
@@ -56,15 +43,16 @@ func IndexDirTree(dir string, rules []Rule, progress chan<- string) error {
 
 func run(zardir string, rules []Rule, progress chan<- string) error {
 	logPath := ZarDirToLog(zardir)
-	var indexers []Indexer
+	var writers []zbuf.WriteCloser
 	for _, rule := range rules {
-		indexer, err := rule.NewIndexer(zardir)
+		w, err := rule.NewIndexer(zardir)
 		if err != nil {
 			return err
 		}
-		indexers = append(indexers, indexer)
+		writers = append(writers, w)
 		if progress != nil {
-			progress <- fmt.Sprintf("%s: creating index %s", logPath, indexer.Path())
+			//XXX
+			progress <- fmt.Sprintf("%s: creating index", logPath)
```

Comment:
Why the XXX?

Reply:
It used to print the name of the index file too but the writer doesn't have a String()

Reply:
It felt right to add back a Path() method to Rule so this addresses it.